src/Command/EditLicensesCommand.php

```diff
@@ -0,0 +1,129 @@
+                'help' => 'Name of the file that contains the sentence ids whose ' .
+                          'license should be changed. ("stdin" will read from ' .
+                          'standard input.)',
+                'required' => true,
+            ])
+            ->addArgument('license', [
+                'help' => 'New license.',
+                'choices' => array_keys(Licenses::getSentenceLicenses())
+            ])
+            ->addOption('dry-run', [
+                'help' => 'Do everything except saving any changes.',
+                'short' => 'n',
+                'boolean' => true
+            ]);
+        return $parser;
+    }
+
+    protected function editLicense($ids, $options) {
+        extract($options);
+        foreach ($ids as $id) {
+            $this->Sentences->getConnection()->transactional(
+                function ($conn) use ($id, $newLicense, $dryRun) {
+                    try {
+                        $sentence = $this->Sentences->get($id);
+                    } catch (RecordNotFoundException $e) {
+                        $this->log[] = ["$id ignored: Record not found"];
+                        return false;
+                    }
+                    $oldLicense = $sentence->license;
+                    if ($oldLicense != $newLicense) {
+                        $newSentence = $this->Sentences->patchEntity(
+                            $sentence,
+                            ['license' => $newLicense]
+                        );
+                        if ($newSentence->hasErrors()) {
+                            $this->log[] = ["$id ignored: Cannot change license"];
+                            return false;
+                        } else {
+                            $this->log[] = [$id, $oldLicense, $newLicense];
+                            if (!$dryRun) {
+                                return $this->Sentences->save($newSentence);
+                            }
+                        }
+                    } else {
+                        $this->log[] = ["$id ignored: License is already $oldLicense"];
+                        return true;
+                    }
+                }
+            );
+        }
+    }
+
+    public function execute(Arguments $args, ConsoleIo $io) {
+        $username = $args->getArgument('username');
+        $userId = $this->Users->getIdFromUsername($username);
+        if (!$userId) {
+            $io->error(format('User "{user}" does not exist!', ['user' => $username]));
+            $this->abort();
+        } else {
+            CurrentUser::store($this->Users->get($userId));
+        }
+
+        $input = $args->getArgument('file');
+        if ($input === 'stdin') {
+            $input = 'php://stdin';
+        } elseif ($input && !file_exists($input)) {
+            $io->error(format('{path} does not exist!', ['path' => $input]));
+            $this->abort();
+        }
+
+        $newLicense = $args->getArgument('license') ?? '';
```

Comment:
Why null coalescing here?

Reply:
The command interprets both no third argument and `''` (the empty string) as `licensing issue`.
If no third argument is given, `$newLicense` will be `null` and while we allow `null` in the `license` column of `sentences` we don't allow it in the `text` column of `contributions` where a license change will be logged.

Do you think it is better to also make the license mandatory, i.e. to always run the command as `cake edit_licenses admin stdin ''` instead of `cake edit_licenses admin stdin` when we want to change the license to `licensing issue`?

Reply:
Thanks for clarifying. Yes, I think it would be better to also make the license mandatory. I can see how inadvertently copy-pasting or pressing enter in the middle of typing the command could cause damage. I’m going to merge the pull request now so that it will be on production this week, but free to make this last change directly on the dev branch.

Reply:
The license is now mandatory too.

And after thinking more about it, using the empty string for a license isn't a good choice, so I've changed the command to accept only the license names we use in the UI.